### PR TITLE
Get the shop members publication tightened up on what is exposed. Fixes reactioncommerce/reaction#399

### PR DESCRIPTION
**Pull request by queso**
Fixes https://github.com/reactioncommerce/reaction/issues/399

### CODE REVIEW

**Comment by queso** [comment]
@aaronjudd you mentioned in chat that you have refactored some of this on a branch? Which branch?

**Comment by aaronjudd** [comment]
the`feature-cart-278` branch: see: https://github.com/reactioncommerce/reaction-core/blob/feature-cart-278/server/publications.coffee

**Comment by queso** [comment]
@aaronjudd I see your refactor and the new way certainly seems ok for what you need to accomplish. I still think there is 0 need to pass things like services data across the wire, so the find call should limit what is being published.

I am happy to move my testing code over to your branch and I can show you what I mean.

File: server/publications.coffee [hunk 1]
```diff
@@ -128,7 +128,7 @@ Meteor.publish 'ShopMembers', ->
   permissions = ['dashboard/orders','owner','admin','dashboard/customers']
   shopId = ReactionCore.getShopId(@)
   if Roles.userIsInRole(@userId, permissions, shopId)
-    return Meteor.users.find()
+    return Meteor.users.find({}, {fields: {_id: 1, email: 1, username: 1, roles: 1}})
```
Comment (queso) [inline server/publications.coffee]: @aaronjudd I think the absence of a fields option being passed is a bad thing... Whenever you do a users.find call, it should be limited to what you need. While I will agree that the code we see now seems secure, what happens if I bug crops up in alanning:roles or in the getShopId code? Let's limit potential exposure by being more exacting on passing what we need.

Comment (aaronjudd) [inline server/publications.coffee]: agreed, make sense

**Pull request merged**